### PR TITLE
Arm64 missing libatomic

### PR DESCRIPTION
**Pull request by karczex**

### CODE REVIEW

**Comment by pawsten** [comment]
Is this PR ready?

File: libebpfdiscovery/headers/ebpfdiscovery/Json.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+namespace boost::json {
+template <typename T>
+void tag_invoke(json::value_from_tag, json::value& v, std::reference_wrapper<T> const& wrapped) {
+	v = json::value_from(wrapped.get());
+}
+} // namespace boost::json
+
+namespace boost::json::ext {
+
+void print(std::ostream& os, boost::json::value const& jv) {
```
Comment (marudno) [inline libebpfdiscovery/headers/ebpfdiscovery/Json.h:32]: consider moving const at the beginning: "const boost::json::value&"

File: libebpfdiscovery/headers/ebpfdiscovery/Json.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	switch (jv.kind()) {
+	case json::kind::object: {
+		auto const& obj = jv.get_object();
```
Comment (marudno) [inline libebpfdiscovery/headers/ebpfdiscovery/Json.h:35]: const auto&?

File: libebpfdiscovery/headers/ebpfdiscovery/Json.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		if (!obj.empty()) {
+			os << "{";
+			for (auto it = obj.begin(); it != obj.end(); ++it) {
+				auto val = it->value();
```
Comment (marudno) [inline libebpfdiscovery/headers/ebpfdiscovery/Json.h:39]: const auto&?

File: libebpfdiscovery/headers/ebpfdiscovery/Json.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			os << "}";
+		}
+		break;
+	}
+
+	case json::kind::array: {
+		auto const& arr = jv.get_array();
+		if (!arr.empty()) {
+			os << "[";
+			auto it = arr.begin();
```
Comment (marudno) [inline libebpfdiscovery/headers/ebpfdiscovery/Json.h:58]: not needed

File: libebpfdiscovery/test/JsonTest.cpp [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+	boost::system::error_code ec;
+	boost::json::parse(json_string, ec);
+	if (ec) {
+		return false;
+	}
+	return true;
+}
+
+TEST_F(JsonTest, removeEmptyKeys) {
+
```
Comment (marudno) [inline libebpfdiscovery/test/JsonTest.cpp:48]: remove redundant empty lines in this file

File: libebpfdiscovery/test/JsonTest.cpp [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+	std::vector<testClass> vtc(4, {"bar", ""});
```
Comment (marudno) [inline libebpfdiscovery/test/JsonTest.cpp:49]: const?

File: libebpfdiscovery/test/JsonTest.cpp [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+	boost::json::object json{{"key", boost::json::value_from(vtc)}};
```
Comment (marudno) [inline libebpfdiscovery/test/JsonTest.cpp:50]: const?

File: libservice/headers/service/Aggregator.h [hunk 8]
```diff
@@ -57,16 +57,14 @@ class Aggregator {
...
-	bool getEnableNetworkCounters() const;
 
 protected:
 	virtual std::chrono::time_point<std::chrono::steady_clock> getCurrentTime() const;
 
 private:
 	const IpAddressChecker& ipChecker;
 
-	std::mutex servicesMutex{};
+	mutable std::mutex servicesMutex{};
```
Comment (marudno) [inline libservice/headers/service/Aggregator.h:67]: remove "mutable"

File: libservice/headers/service/Service.h [hunk 9]
```diff
@@ -39,23 +41,50 @@ struct ArrayHasher {
...
 
-} // namespace service
\ No newline at end of file
+// cppcheck-suppress unknownMacro
+BOOST_DESCRIBE_STRUCT(Service, (), (pid, endpoint, domain, scheme, internalClientsNumber, externalClientsNumber, externalIPv4_16ClientNets, externalIPv4_24ClientNets, externalIPv6ClientsNets))
+
+} // namespace service
+
+namespace boost::json {
+
```
Comment (marudno) [inline libservice/headers/service/Service.h:73]: remove redundant lines

File: libservice/headers/service/Service.h [hunk 10]
```diff
@@ -39,23 +41,50 @@ struct ArrayHasher {
...
+ 	  inline void tag_invoke(value_from_tag, value& v, service::Service::IPv6NetworksMap_t map) {
+			if(map.empty()){
+				v = value{};
+			} else {
+	      v = json::value_from(map.size());
+	    }
+	  }
+
+
+		inline void tag_invoke(json::value_from_tag, json::value& v, service::Service::IPv4NetworksMap_t map) {
```
Comment (marudno) [inline libservice/headers/service/Service.h:83]: reformat file
service::Service::IPv4NetworksMap_t by const&

File: libservice/headers/service/Service.h [hunk 11]
```diff
@@ -39,23 +41,50 @@ struct ArrayHasher {
...
-} // namespace service
\ No newline at end of file
+// cppcheck-suppress unknownMacro
+BOOST_DESCRIBE_STRUCT(Service, (), (pid, endpoint, domain, scheme, internalClientsNumber, externalClientsNumber, externalIPv4_16ClientNets, externalIPv4_24ClientNets, externalIPv6ClientsNets))
+
+} // namespace service
+
+namespace boost::json {
+	
+ 	  inline void tag_invoke(value_from_tag, value& v, service::Service::IPv6NetworksMap_t map) {
```
Comment (marudno) [inline libservice/headers/service/Service.h:74]: const service::Service::IPv6NetworksMap_t&

File: ebpfdiscoverysrv/src/main.cpp [hunk 12]
```diff
@@ -22,6 +22,7 @@
 #include <boost/program_options.hpp>
 
 #include <atomic>
+#include <mutex>
```
[the remark this answers is not in the document]
Comment (karczex) [reply ebpfdiscoverysrv/src/main.cpp:25]: yes, it's used in line 40

File: libebpfdiscovery/test/JsonTest.cpp (hunk 6 above)
Comment (karczex) [reply libebpfdiscovery/test/JsonTest.cpp:48]: Not necessarily

**Review by filip-szweda — DISMISSED** [review]

**Review by filip-szweda — APPROVED** [review]

**Pull request merged**